### ISSUE TITLE
Constraint and required messages are almost invisible on Android 8.1

### ISSUE THREAD

**Issue by tinok** [getodk/collect#2100]
#### Software and hardware versions 
Collect v1.14.0, Android v8.1.0, Nexus 5X

#### Problem description
Constraint and required messages appear on a very light background with white font, making them close to invisible. See screenshot:

![image](https://user-images.githubusercontent.com/673511/38439968-a590d9c4-39ad-11e8-83fc-783656a48ccd.png)

**Comment by lognaturel** [comment]
Can't reproduce on android 5 and 6 so this is probably an Android 8 problem related to [the change to white toasts](https://www.androidpolice.com/2017/10/26/android-8-1-feature-spotlight-toast-messages-now-white-better-match-rest-ui/).

**Comment by huangyz0918** [comment]
Interesting, at first I simply thought it is just a normal phenomenon. I can definitely let the toast change color to dark when I change a wallpaper which is dark color. Android 8.1 can automatically change the system theme according to your wallpaper color, as well as the toast color. So we can use a fixed color to cover all the APIs in `toast_view.xml`. @opendatakit-bot claim

**Comment by lognaturel** [comment]
@huangyz0918 an ideal solution would let Android 8.1 behave as it does by default. I haven't had a chance to look yet but I suspect something is forcing toast text color to white when it should be determined by the system.

**Comment by lognaturel** [comment]
<strike>Indeed the issue is with the `toast_view` layout which forces white text color. @shobhitagarwal1612 actually ended up addressing this in #2061 by removing the redundant `showCustomToast` method but</strike> 

Actually, the custom layout does several useful things like increasing the font size and positioning the toast above the keyboard. So @grzesiek2010 you were very right that the change to replace `showCustomToast` was too significant to sneak into #2061. @shobhitagarwal1612 please make sure that is reverted.

The fix @huangyz0918 proposes in #2103 seems like the best we can do. Ideally we could let the system determine the background and the font but I see no way to access the toast font color.

**Comment by huangyz0918** [comment]
We can change toast color (including text color) in Java code, just like this:

``` java
If (API level > 26) {
    Toast toast = Toast.makeText(this, resId, Toast.LENGTH_SHORT);
    Textview textview = toast.getView().findViewById(id);
    textview.setBackgroundColor(Excepted color);
    textview.setTextColor(Expected color);
}
```
 Add these code to our custom toast controller. I haven’t try it, but I think it is also ok.
